pkg/gitops/flux/installer.go

```diff
@@ -133,6 +135,28 @@ func NewInstaller(ctx context.Context, k8sRestConfig *rest.Config, k8sClientSet
 
 // Run runs the Flux installer
 func (fi *Installer) Run(ctx context.Context) error {
+	fluxNSExists, err := kubernetes.CheckNamespaceExists(fi.k8sClientSet, fi.opts.Namespace)
+	if err != nil {
+		return errors.Wrapf(err, "cannot check if namespace %s exists", fi.opts.Namespace)
+	}
+	if fluxNSExists {
+		logger.Info("Namespace \"%s\" already exists. eksctl will now delete all its content before proceeding to install Flux.", fi.opts.Namespace)
+		cont, err := continueInstallation()
+		if err != nil {
+			return err
+		}
+		if !cont {
+			logger.Info("Installation of Flux was aborted.")
+			return nil
+		}
+		if cont {
+			err := kubernetes.DeleteNamespace(fi.k8sClientSet, fi.opts.Namespace)
```

Comment:
Shouldn't we also be deleting the roles and helmrelease CRD?

Reply:
We could, though in practice, there is no conflict for these, given their simplicity, hence I didn't bother. Indeed `CreateOrUpdate` deals with them perfectly well:
```console
[ℹ]  replaced "CustomResourceDefinition.apiextensions.k8s.io/helmreleases.helm.fluxcd.io"
[...]
[ℹ]  replaced "ClusterRoleBinding.rbac.authorization.k8s.io/tiller"
[...]
[ℹ]  replaced "kube-system:RoleBinding.rbac.authorization.k8s.io/tiller-user-binding"
[...]
[ℹ]  replaced "ClusterRole.rbac.authorization.k8s.io/flux"
[ℹ]  replaced "ClusterRoleBinding.rbac.authorization.k8s.io/flux"
[...]
[ℹ]  replaced "ClusterRole.rbac.authorization.k8s.io/flux-helm-operator"
[ℹ]  replaced "ClusterRoleBinding.rbac.authorization.k8s.io/flux-helm-operator"
```

Reply:
This bring the topic of "what constitutes a component?". Proper way to do it would be via [Application CRD](https://github.com/kubernetes-sigs/application), I strongly recommend that we adopt it going forward.